File: gto-support-room/src/main/java/org/ccci/gto/android/common/room/converter/LocaleConverter.java
package org.ccci.gto.android.common.room.converter;

import org.ccci.gto.android.common.compat.util.LocaleCompat;

import java.util.Locale;

import androidx.annotation.Nullable;
import androidx.room.TypeConverter;

public final class LocaleConverter {
    @Nullable
    @TypeConverter
    public static Locale toLocale(@Nullable final String tag) {
        return tag != null ? LocaleCompat.forLanguageTag(tag) : null;
    }

    @Nullable
    @TypeConverter
    public static String toString(@Nullable final Locale locale) {
        return locale != null ? LocaleCompat.toLanguageTag(locale) : null;
    }
}

File: gto-support-room/src/main/java/org/ccci/gto/android/common/room/converter/LocaleConverter.kt
package org.ccci.gto.android.common.room.converter

import androidx.room.TypeConverter
import java.util.Locale
import org.ccci.gto.android.common.compat.util.LocaleCompat

@Deprecated("Since v3.4.0, use converter from the gto-support-androidx-room module instead")
object LocaleConverter {
    @JvmStatic
    @TypeConverter
    fun toLocale(tag: String?) = tag?.let { LocaleCompat.forLanguageTag(it) }

    @JvmStatic
    @TypeConverter
    fun toString(locale: Locale?) = locale?.let { LocaleCompat.toLanguageTag(it) }
}
